### PR TITLE
[4.1] Add dragula autoscroll

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes

Add Auto scroll in the draggable lists, eg drag an item and the list will automatically scroll up/down if you reached the bottom/top of the list.

### Testing Instructions

Apply the PR and check the `administrator/index.php?option=com_plugins&view=plugins` to see if the list auto scrolls when the dragged item reaches to the bottom of the page

### CODE REVIEW

**Comment by brianteeman** [comment]
nice find

File: build/build-modules-js/settings.json [hunk 1]
```diff
@@ -341,6 +341,25 @@
         "dependencies": [],
         "licenseFilename": "license"
       },
+      "dom-autoscroller": {
+        "name": "dragula-scroller",
+        "js": {
+          "dist/dom-autoscroller.js": "js/dragula-scroller.js",
```
Comment (brianteeman) [inline build/build-modules-js/settings.json]: ?? is it good practice to rename the script?

Comment (dgrammatiko) [reply build/build-modules-js/settings.json]: Sure, the build script is allowing this so the project names things as it finds more appropriate. I mean if the core provides some script then devs should use that version. If they want to roll their own dependencies that's also fine as long as none of their pages doesn't load some scripts from the core and the exact same from their own distribution. In short, this shouldn't be a problem if devs are not mixing the same script from different folders (the naming won't help in such case, they're destined to fail either way)

Comment (dgrammatiko) [reply build/build-modules-js/settings.json]: Never mind, I already renamed the files to match the source

**Comment by PhocaCz** [comment]
I have tested this item :red_circle: unsuccessfully on e8773893886b6ca01e42dae8bd874d8ee2268263<br /><br />Hi,

it does work for standard items (e.g. articles) but e.g. on Linux, the scrolling starts only in specific position, very hard to find it - in fact it is very hard to start the autoscroll


https://i.imgur.com/GOqpBcL.gif

In subforms, it does not work entirely.

Jan<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36354">issues.joomla.org/tracker/joomla-cms/36354</a>.</sub>

**Comment by dgrammatiko** [comment]
> In subforms, it does not work entirely.

Subforms are NOT using the same drag and drop script, so that's expected

**Comment by PhocaCz** [comment]
>Subforms are NOT using the same drag and drop script, so that's expected

Ok, as you can see on my GIF, I get problems to sort on my Linux, so will be good, if other can test it in other environments.

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on e8773893886b6ca01e42dae8bd874d8ee2268263<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36354">issues.joomla.org/tracker/joomla-cms/36354</a>.</sub>

**Comment by brianteeman** [comment]
The drag scroll is very very slow
It is not longer possible to drop except at the last position<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36354">issues.joomla.org/tracker/joomla-cms/36354</a>.</sub>

**Comment by brianteeman** [comment]
Note: i tested the codepen which worked fine

**Comment by PhocaCz** [comment]
> The drag scroll is very very slow

I have the same behaviour on Linux (KDE), it starts to be faster but only in specific position and it is hard to find this position.

**Comment by PhocaCz** [comment]
>let's use the max speed param

For me, there is no change :-( ... it is still slow until I find randomly the place where it starts to work (but as written above, it is really hard to find this place)

**Comment by dgrammatiko** [comment]
I'm putting this in draft mode. Will revisit later

**Comment by dgrammatiko** [comment]
Closing here as the solution is different than the approach in this PR

**Pull request closed without merging**